chia/rpc/wallet_rpc_api.py

```diff
@@ -4496,3 +4504,83 @@ class CRCATApprovePending(Streamable):
         return {
             "transactions": [tx.to_json_dict_convenience(self.service.config) for tx in txs],
         }
+
+    ##########################################################################################
+    # VAULT
+    ##########################################################################################
+    @tx_endpoint(push=True)
+    async def vault_create(
+        self,
+        request: Dict[str, Any],
+        tx_config: TXConfig = DEFAULT_TX_CONFIG,
+        extra_conditions: Tuple[Condition, ...] = tuple(),
+    ) -> EndpointResult:
+        """
+        Create a new vault
+        """
+        assert self.service.wallet_state_manager
+        wallet = self.service.wallet_state_manager.main_wallet
+
+        secp_pk = bytes.fromhex(str(request.get("secp_pk")))
+        entropy = bytes.fromhex(str(request.get("entropy")))
+        bls_pk = G1Element.from_bytes(bytes.fromhex(str(request.get("bls_pk"))))
+        timelock = uint64(request["timelock"])
+        fee = request.get("fee", 0)
+        vault_puzzlehash = get_vault_puzzle_hash(
+            secp_pk, DEFAULT_CONSTANTS.GENESIS_CHALLENGE, entropy, bls_pk, timelock
+        )
+
+        # Get xch coin
+        amount = uint64(1)
+        coins = await wallet.select_coins(uint64(amount + fee), tx_config.coin_selection_config)
+
+        # Create singleton launcher
+        origin = coins.copy().pop()
+        launcher_coin = Coin(origin.name(), SINGLETON_LAUNCHER_PUZZLE_HASH, amount)
+
+        vault_full_puzzlehash = create_singleton_puzzle_hash(vault_puzzlehash, launcher_coin.name())
+        announcement_message = Program.to([vault_puzzlehash, amount, bytes(0x80)]).get_tree_hash()
```

Comment:
We should hint the secp_pk and hidden_puzzle_hash here so the information is not entirely lost to the chain